compiler/src/test/java/io/neow3j/compiler/SafeMethodTest.java

```diff
@@ -0,0 +1,39 @@
+package io.neow3j.compiler;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import io.neow3j.devpack.annotations.Safe;
+import io.neow3j.protocol.core.methods.response.ContractManifest.ContractABI.ContractMethod;
+import java.io.IOException;
+import java.util.List;
+import org.junit.Test;
+
+public class SafeMethodTest {
+
+    @Test
+    public void methodShouldBeMarkedSafeIfAnnotatedWithSafeAnnotation() throws IOException {
+        CompilationUnit unit = new Compiler().compileClass(SafeMethodTestContract.class.getName());
+        List<ContractMethod> methods = unit.getManifest().getAbi().getMethods();
+        assertTrue(methods.get(0).isSafe());
+    }
+
+    @Test
+    public void methodShouldNotBeMarkedSafeIfNotAnnotatedWithSafeAnnotation() throws IOException {
+        CompilationUnit unit = new Compiler().compileClass(SafeMethodTestContract.class.getName());
+        List<ContractMethod> methods = unit.getManifest().getAbi().getMethods();
+        assertFalse(methods.get(1).isSafe());
+    }
+
+    static class SafeMethodTestContract {
```

Comment:
@csmuller just for the sake of completeness: can you add a test case where a contract has at least two `@Safe` methods? 😄 

Moreover, is there any restriction for this `@Safe` annotation? Like, for example, all other annotations can be added if `@Safe` is present, right?

Reply:
> just for the sake of completeness: can you add a test case where a contract has at least two @Safe methods?

Will do.

> Moreover, is there any restriction for this @Safe annotation? Like, for example, all other annotations can be added if @Safe is present, right?

Good thinking. I found one point. It doesn't make sense to use the annotation on methods that are not public, i.e. that  don't show up in the manifest. I will throw an exception if the annotation is used on non-public methods.

I don't see conflicts with other annotations like `OnDeployment` or `OnVerification`. Just because methods marked with such annotations are special methods it doesn't mean that the developer cannot mark them as "safe".